File: samples/VisualBasic/Northwind/CustomerDemographic.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="CustomerDemographics")>
   Public Class CustomerDemographic

      <Column>
      Property CustomerDesc As String

      <Column(IsPrimaryKey:=True)>
      Property CustomerTypeID As String

      <Association(OtherKey:=NameOf(CustomerCustomerDemo.CustomerTypeID))>
      ReadOnly Property CustomerCustomerDemos As New Collection(Of CustomerCustomerDemo)

   End Class

End Namespace

File: samples/CSharp/Northwind/CustomerDemographic.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "CustomerDemographics")]
   public class CustomerDemographic {

      [Column(IsPrimaryKey = true)]
      public string CustomerTypeID { get; set; }

      [Column]
      public string CustomerDesc { get; set; }

      [Association(OtherKey = nameof(CustomerCustomerDemo.CustomerTypeID))]
      public Collection<CustomerCustomerDemo> CustomerCustomerDemos { get; } = new Collection<CustomerCustomerDemo>();
   }
}
